hatchet_sdk/loader.py

```diff
@@ -1,246 +1,147 @@
-            otel_exporter_oltp_headers=otel_exporter_oltp_headers,
-            otel_exporter_oltp_protocol=otel_exporter_oltp_protocol,
-            worker_healthcheck_port=worker_healthcheck_port,
-            worker_healthcheck_enabled=worker_healthcheck_enabled,
-        )
-
-    def _load_tls_config(self, tls_data: Dict, host_port) -> ClientTLSConfig:
-        tls_strategy = (
-            tls_data["tlsStrategy"]
-            if "tlsStrategy" in tls_data
-            else self._get_env_var("HATCHET_CLIENT_TLS_STRATEGY")
-        )
-
-        if not tls_strategy:
-            tls_strategy = "tls"
-
-        cert_file = (
-            tls_data["tlsCertFile"]
-            if "tlsCertFile" in tls_data
-            else self._get_env_var("HATCHET_CLIENT_TLS_CERT_FILE")
-        )
-        key_file = (
-            tls_data["tlsKeyFile"]
-            if "tlsKeyFile" in tls_data
-            else self._get_env_var("HATCHET_CLIENT_TLS_KEY_FILE")
-        )
-        ca_file = (
-            tls_data["tlsRootCAFile"]
-            if "tlsRootCAFile" in tls_data
-            else self._get_env_var("HATCHET_CLIENT_TLS_ROOT_CA_FILE")
-        )
-
-        server_name = (
-            tls_data["tlsServerName"]
-            if "tlsServerName" in tls_data
-            else self._get_env_var("HATCHET_CLIENT_TLS_SERVER_NAME")
-        )
-
-        # if server_name is not set, use the host from the host_port
-        if not server_name:
-            server_name = host_port.split(":")[0]
-
-        return ClientTLSConfig(tls_strategy, cert_file, key_file, ca_file, server_name)
-
-    @staticmethod
-    def _get_env_var(env_var: str, default: Optional[str] = None) -> str:
-        return os.environ.get(env_var, default)
+            if not token:
+                raise ValueError("Either the token or tenant_id must be set")
+
+            return get_tenant_id_from_jwt(token)
+
+        return tenant_id
+
+    @field_validator("host_port", mode="after")
+    @classmethod
+    def validate_host_port(cls, host_port: str, info: ValidationInfo) -> str:
+        token = cast(str, info.data.get("token"))
+
+        _, grpc_broadcast_address = get_addresses_from_jwt(token)
+
+        return grpc_broadcast_address
+
+    @field_validator("server_url", mode="after")
+    @classmethod
+    def validate_server_url(cls, server_url: str, info: ValidationInfo) -> str:
+        token = cast(str, info.data.get("token"))
+
+        _server_url, _ = get_addresses_from_jwt(token)
+
+        return _server_url
+
+    @field_validator("tls_config", mode="after")
+    @classmethod
+    def validate_tls_config(
+        cls, tls_config: ClientTLSConfig, info: ValidationInfo
+    ) -> ClientTLSConfig:
+        host_port = cast(str, info.data.get("host_port"))
+
+        return _load_tls_config(host_port)
+
+    def __hash__(self) -> int:
+        return hash(json.dumps(self.model_dump(), default=str))
+
+    ## TODO: Fix host port overrides here
```

Comment:
address todo?

Reply:
I was planning to leave this for now since it's related to [that outstanding PR in the non-fork](https://github.com/hatchet-dev/hatchet-python/pull/243) - happy to address this now though!

Reply:
this is done 👍